packaging/os/yumrepo.py

```diff
@@ -0,0 +1,529 @@
+        SI prefix (k, M or G) may be appended to the bandwidth value.
+  timeout:
+    required: false
+    default: 30
+    description:
+      - Number of seconds to wait for a connection before timing out.
+  username:
+    required: false
+    default: None
+    description:
+      - Username to use for basic authentication to a repo or really any url.
+
+extends_documentation_fragment: files
+
+notes:
+  - All comments will be removed if modifying an existing repo file.
+  - Section order is preserved in an existing repo file.
+  - Parameters in a section are ordered alphabetically in an existing repo
+    file.
+  - The repo file will be automatically deleted if it contains no repository.
+'''
+
+EXAMPLES = '''
+- name: Add repository
+  yumrepo:
+    name: epel
+    description: EPEL YUM repo
+    baseurl: http://download.fedoraproject.org/pub/epel/$releasever/$basearch/
+
+- name: Add multiple repositories into the same file (1/2)
+  yumrepo:
+    name: epel
+    description: EPEL YUM repo
+    file: external_repos
+    baseurl: http://download.fedoraproject.org/pub/epel/$releasever/$basearch/
+    gpgcheck: no
+- name: Add multiple repositories into the same file (2/2)
+  yumrepo:
+    name: rpmforge
+    description: RPMforge YUM repo
+    file: external_repos
+    baseurl: http://apt.sw.be/redhat/el7/en/$basearch/rpmforge
+    mirrorlist: http://mirrorlist.repoforge.org/el7/mirrors-rpmforge
+    enabled: no
+
+- name: Remove repository
+  yumrepo:
+    name: epel
+    state: absent
+
+- name: Remove repository from a specific repo file
+  yumrepo:
+    name: epel
+    file: external_repos
+    state: absent
+'''
+
+
+class YumRepo(object):
+    # Class global variables
+    module = None
+    params = None
+    section = None
+    repofile = ConfigParser.RawConfigParser()
+
+    # List of parameters which can be a list
+    list_params = ['exclude', 'includepkgs']
+
+    # List of keys which will be excluded from the final output
+    hidden_params = [
+        'file', 'repoid', 'reposdir', 'state', 'dest', 'checkmode',
+        'mode', 'owner', 'group', 'seuser', 'serole', 'setype', 'selevel']
+
+    def __init__(self, module):
+        # To be albe to use fail_json
+        self.module = module
+        # Shortcut for the params
+        self.params = self.module.params
+        # Section is always the repoid
+        self.section = self.params['repoid']
+
+        # Check if repo directory exists
+        repos_dir = self.params['reposdir']
+        if not os.path.isdir(repos_dir):
+            self.module.fail_json(
+                msg='Repo directory "%s" does not exist.' % repos_dir)
+
+        # Get the given or the default repo file name
+        repo_file = self.params['repoid']
+        if self.params['file'] is not None:
+            repo_file = self.params['file']
+
+        # Set dest; also used to set dest parameter for the FS attributes
+        self.params['dest'] = "%s/%s.repo" % (repos_dir, repo_file)
```

Comment:
Why not `os.path.join(repos_dir, repo_file)`?  Would avoid double slashes if the user specifies `repos_dir=/with/a/trailing/slash/`.